docs/angular/your-first-app/4-loading-photos.md

```diff
@@ -1,69 +1,214 @@
+Next, at the end of the `addNewToGallery()` method, add a call to `Preferences.set()` to save the `photos` array. By adding it here, the `photos` array is stored each time a new photo is taken. This way, it doesn’t matter when the app user closes or switches to a different app - all photo data is saved.
+
+```ts
+public async addNewToGallery() {
+  // Take a photo
+  const capturedPhoto = await Camera.getPhoto({
+    resultType: CameraResultType.Uri,
+    source: CameraSource.Camera,
+    quality: 100,
+  });
 
-```tsx
-Preferences.set({
-  key: this.PHOTO_STORAGE,
-  value: JSON.stringify(this.photos),
-});
+  const savedImageFile = await this.savePicture(capturedPhoto);
+
+  this.photos.unshift(savedImageFile);
+
+  // CHANGE: Add method to cache all photo data for future retrieval.
+  Preferences.set({
+    key: this.PHOTO_STORAGE,
+    value: JSON.stringify(this.photos),
+  });
+}
 ```
 
-With the photo array data saved, create a function called `loadSaved()` that can retrieve that data. We use the same key to retrieve the photos array in JSON format, then parse it into an array:
+With the photo array data saved, create a new public method in the `PhotoService` class called `loadSaved()` that can retrieve the photo data. We use the same key to retrieve the `photos` array in JSON format, then parse it into an array:
 
-```tsx
-public async loadSaved() {
-  // Retrieve cached photo array data
-  const { value } = await Preferences.get({ key: this.PHOTO_STORAGE });
-  this.photos = (value ? JSON.parse(value) : []) as UserPhoto[];
+```ts
+export class PhotoService {
+  // Same old code from before.
+
+  // CHANGE: Add the method to load the photo data.
+  public async loadSaved() {
+    // Retrieve cached photo array data
+    const { value: photoList } = await Preferences.get({ key: this.PHOTO_STORAGE });
+    this.photos = (photoList ? JSON.parse(photoList) : []) as UserPhoto[];
+  }
+}
+```
 
-  // more to come...
+On mobile (coming up next!), we can directly set the source of an image tag - `<img src="x" />` - to each photo file on the `Filesystem`, displaying them automatically. On the web, however, we must read each image from the `Filesystem` into base64 format, using a new `base64` property on the `Photo` object. This is because the `Filesystem` API uses [IndexedDB](https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API) under the hood. Add the following code to complete the `loadSaved()` method:
+
+```ts
+export class PhotoService {
+  // Same old code from before.
+
+  // CHANGE: Update the `loadSaved` method.
+  public async loadSaved() {
+    // Retrieve cached photo array data
+    const { value: photoList } = await Preferences.get({ key: this.PHOTO_STORAGE });
+    this.photos = (photoList ? JSON.parse(photoList) : []) as UserPhoto[];
+
+    // CHANGE: Display the photo by reading into base64 format.
```

Comment:
```suggestion
    // CHANGE: Display the photo by reading into base64 format
```

Can we be consistent about periods on the end of comments?

Reply:
Do you prefer with or without?